GEMM benchmark with ARM Compute Library

Nyrio opened the pull request:
Mostly done but need to add the verification, which requires #137

mehdi-goli commented on benchmark/acl/CMakeLists.txt:
```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.2.2)
```
why 3.2.2?

Nyrio replied:
Because we have that in the other benchmark cmake files as well. I don't know which version I should require so unless a cmake pro (e.g @mmha) suggests something else, I will use the same version than in the other similar files for consistency.

mehdi-goli replied:
3.4.2

Nyrio replied:
Ok, fixed for all the benchmarks

mehdi-goli commented on benchmark/acl/blas3/gemm.cpp:
```diff
@@ -0,0 +1,163 @@
+  };
+
+  // Warm up to avoid benchmarking data transfer
+  // blas_benchmark::utils::warmup(blas_method_def);
```
Is this deliberate to be commented out?

Nyrio replied:
It was because the function was moved to the `common_utils` in another MR, now that it's updated on top of everything I enabled the warmup

Nyrio commented:
I've added the verification to the ACL benchmarks as well.
Also I've added a condition to show an error (a red line like verification failures) when the user attempts to use transposed matrices with this library. These benchmarks only support `n,n` mode.
